microsoft/testsuites/network/common.py

```diff
@@ -26,7 +27,10 @@ def initialize_nic_info(
             assert_that(sriov_count).described_as(
                 f"there is no sriov nic attached to VM {node.name}"
             ).is_greater_than(0)
-        nics_info = Nics(node)
+        if not isinstance(node.os, BSD):
```

Comment:
This should be wrapped into nic.py. Use a method like `create_nics`

Reply:
Sure! `Nics` class is also used in bunch of other places and I am testing if replacing those won't have any regression in existing testcases